File: XperiaModules/Touch/XperiaTouchSecTS/src/com/xperia/settings/touch/BootCompletedReceiver.java
/*
 * Copyright (C) 2024 XperiaLabs Project
 * Copyright (C) 2018 The LineageOS Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.xperia.settings.touch;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.hardware.display.DisplayManager;
import android.os.IBinder;
import android.util.Log;
import android.view.Display;

import com.xperia.settings.touch.polling.HighTouchPollingService;

public class BootCompletedReceiver extends BroadcastReceiver {

    private static final String TAG = "XperiaTouch";

    @Override
    public void onReceive(final Context context, Intent intent) {
        Log.i(TAG, "Received intent: " + intent.getAction());

        switch (intent.getAction()) {
            case Intent.ACTION_LOCKED_BOOT_COMPLETED:
                onLockedBootCompleted(context);
                break;
        }
    }

    private static void onLockedBootCompleted(Context context) {
        // Services that don't require reading from data.
        HighTouchPollingService.startService(context);
    }
}

File: XperiaModules/SIM/XperiaEuicc/src/com/xperia/settings/euicc/BootCompletedReceiver.kt
/*
 * Copyright (C) 2024 XperiaLabs Project
 * Copyright (C) 2021 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package com.xperia.settings.euicc

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BootCompletedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "Received boot completed intent")
        EuiccDisabler.enableOrDisableEuicc(context)
    }

    companion object {
        private const val TAG = "XperiaEuicc"
    }
}
